lib/AST/Decl.cpp

```diff
@@ -9464,6 +9464,19 @@ GenericTypeParamDecl *OpaqueTypeDecl::getExplicitGenericParam(
   return genericParamType->getDecl();
 }
 
+bool OpaqueTypeDecl::exportUnderylingTypeToClients() const {
+  auto mod = getDeclContext()->getParentModule();
+  if (mod->getResilienceStrategy() != ResilienceStrategy::Resilient)
+    return true;
+
+  ValueDecl *namingDecl = getNamingDecl();
+  if (namingDecl->getAttrs().hasAttribute<AlwaysEmitIntoClientAttr>() ||
+      namingDecl->getAttrs().hasAttribute<InlinableAttr>())
```

Comment:
@tshortli Should we also consider back deployed functions as inalienable in such a case?